lib/Ast.ml

```diff
@@ -2286,6 +2286,11 @@ end = struct
     | Exp {pexp_desc= Pexp_field (e, _); _}, {pexp_desc= Pexp_construct _; _}
       when e == exp ->
         true
+    | ( Exp {pexp_desc= Pexp_apply (op, [_; (Nolabel, e)]); _}
+      , {pexp_desc= Pexp_list _; pexp_attributes= _ :: _; _} )
+      when Exp.is_infix op && e == exp ->
+        true
+    | _, {pexp_desc= Pexp_list _; _} -> false
```

Comment:
This case seems too easy and shortcuts the next case, which checks a lot of things. Here is an example where the AST changes:
```ocaml
let _ = f 1 ([e; f] [@a])
```
I think this case should be moved to after the next one, which already does what we expect (except in the local open case). I'm still not 100% confident this will be ok in every cases.

Reply:
I fixed this case and made sure we don't break the parenze_exp function